plumbing/cst/chain_state.go

```diff
@@ -75,7 +77,13 @@ func (chn *ChainStateProvider) GetBlock(ctx context.Context, id cid.Cid) (*types
 
 // SampleRandomness samples randomness from the chain at the given height.
 func (chn *ChainStateProvider) SampleRandomness(ctx context.Context, sampleHeight *types.BlockHeight) ([]byte, error) {
-	tipSetBuffer, err := chain.GetRecentAncestorsOfHeaviestChain(ctx, chn.reader, sampleHeight)
+	head := chn.reader.GetHead()
```

Comment:
These changes appear unrelated to the rest of the changes in this PR, is this an optimization?

Reply:
This inlining GetRecentAncestorsOfHeaviestChain into its only call site narrows the interface of the chain reader passed on to GetRecentAncestors so that it can be satisfied by the new builder (no longer needs to know about head).